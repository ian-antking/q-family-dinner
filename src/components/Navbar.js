import React from "react";
import { Link } from 'react-router-dom';
import styled from "styled-components";

const StyledNavbar = styled.div`
position: fixed;
top: 0;
left: 0;
height: 50px;
width: calc(100vw - 30px);
padding: 0 15px;
color: #000000;
background: whitesmoke;
display: flex;
flex-flow: row nowrap;
justify-content: space-between;
align-items: center;
z-index: 9;

	a {
		font-size: 1.8em;
		font-weight: bold;
		color: #000000;
		text-decoration: none;
	}
`;


const NavBar = () => {
	return (
		<StyledNavbar>
			<Link to="/">Queer Family Tea</Link>
		</StyledNavbar>
	);
};

export default NavBar;

