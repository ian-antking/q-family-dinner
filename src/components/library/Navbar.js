import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import BurgerMenu from './BurgerMenu';

const StyledNavbar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  height: 50px;
  width: calc(100vw - 20px);
  padding: 0 5px 0 15px;
  color: ${(props) => props.theme.navbarText};
  background: ${(props) => props.theme.navbar};
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
  z-index: 9;

  a {
    font-size: 1.8em;
    font-weight: bold;
    color: #000000;
    text-decoration: none;
  }

  @media (min-width: 768px) {
    padding: 0 15px;
  }
`;

const NavBar = ({ content }) => {
  return (
    <StyledNavbar data-testid="navbar">
      <Link data-testid="homeLink" to="/">Queer Family Tea</Link>
      <BurgerMenu content={content}/>
    </StyledNavbar>
  );
};

export default NavBar;
