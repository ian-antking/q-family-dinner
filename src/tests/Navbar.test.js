import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';

test('snapshot: Navbar component', () => {
  const { asFragment } = render(
    <Router>
      <Navbar />
    </Router>
  );
  const component = asFragment();
  expect(component).toMatchSnapshot();
});

test('Navbar component renders', () => {
  const {getByTestId } = render(
    <Router>
      <Navbar />
    </Router>
  );
  const componentId = getByTestId('navbar');
  expect(componentId).toBeInTheDocument();
});

