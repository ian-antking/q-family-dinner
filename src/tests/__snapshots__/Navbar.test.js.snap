// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: Navbar component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iKRehC"
    data-testid="navbar"
  >
    <a
      data-testid="homeLink"
      href="#/"
    >
      Queer Family Tea
    </a>
  </div>
</DocumentFragment>
`;
